### PR TITLE
Remove hourly summation elements if weekly billable project exists on timecard

### PR DESCRIPTION
**Pull request by caleywoods**
## Description

Remove hourly summation elements if weekly billable project exists on timecard

## Additional information

The hourly summation elements in the footer should hide upon selection of a weekly billable project as well as being hidden when coming into a saved timecard that has a weekly billable project on it. Removing any weekly billable projects from the timecard either by deleting it or updating it should restore the hourly summation elements.

JS tests were added and were passing on my local machine but seem to fail when ran inside the docker instance, that probably needs its own issue.

### CODE REVIEW

**Comment by kewlguy781** [comment]
CircleCi show as fail. https://app.circleci.com/pipelines/github/18F/tock/1458/workflows/5e7d1820-5fa8-4ace-8d78-16c3dbe22d5b/jobs/7259

File: tock/tock/static/js/components/timecard.js [hunk 1]
```diff
@@ -363,6 +368,56 @@ function handleConfirm(val) {
...
+ * @name handleBillingElementState
+ * @param {string} visibility
+ * */
+function handleBillingElementState(visibility) {
+  const showBillingHourSummationElements = visibility === 'show';
+  const totalReportedElement = document.querySelector('#total-reported-div');
+  const totalBillableElement = document.querySelector('#total-billable-div');
+
+  if (showBillingHourSummationElements) {
+    const weeklyBillingAlertElement = document.querySelector('#weekly-billing-alert');
```
Comment (neilmb) [inline tock/tock/static/js/components/timecard.js:382]: **suggestion (non-blocking):** these two calls are repeated inside the conditional, but the ones for `totalReportedElement` and `totalBillableElement` are not. I think it's clearer to be consistent and pull all four of the `querySelector` calls outside the if statements and only make the changes in the if and else blocks.

File: tock/tock/static/js/components/timecard.js [hunk 2]
```diff
@@ -363,6 +368,56 @@ function handleConfirm(val) {
...
+  const selects = document.querySelectorAll('.entry-project select');
+  let weeklyBilledProjectSeen = false;
+  for (let i = 0; i < selects.length; i++) {
+    const currentSelect = selects[i];
+    const project = currentSelect.selectedOptions[0];
+    const projectIsWeeklyBilled = (project.dataset || {}).is_weekly_bill === 'true';
+    // Flip our 'state' so we know we've encountered at least one weekly billed project
+    // Get out of the loop as soon as we've found one weekly billed project
+    if (projectIsWeeklyBilled && !weeklyBilledProjectSeen) {
+      weeklyBilledProjectSeen = true;
```
Comment (neilmb) [inline tock/tock/static/js/components/timecard.js:414]: **suggestion (non-blocking):** I think that we could `return true;` directly from here and then change L418 to `return false;`. Then the `weeklyBilledProjectSeen` variable isn't needed at all.

File: tock/tock/tests/js/integration/timecard.test.js [hunk 3]
```diff
@@ -81,7 +81,86 @@ describe("Timecard", () => {
...
+  describe("handles weekly billing", () => {
+    beforeEach(() => page.goto(`${baseUrl}/reporting_period/2015-03-30/`));
+
+    test("hides hours input and shows project allocation element", async () => {
+      // https://github.com/18F/tock/issues/848
+      await page.evaluate(() => {
+        document.querySelector("#id_timecardobjects-0-hours_spent").value = ''
+      });
+
+      // @TODO: Seed database with a weekly billing project and know the ID for this test
```
Comment (neilmb) [inline tock/tock/tests/js/integration/timecard.test.js:95]: **question:** It looks like this is already done in the test data, do we still need this TODO comment?

**Review by neilmb — APPROVED** [review]
Great to have this, just a few comments and minor suggestions.

**Pull request merged**